worlds/cvcotm/rules.py

```diff
@@ -0,0 +1,208 @@
+        return state.has_any([iname.double, iname.roc_wing], self.player)
+
+    def has_jump_level_2(self, state: CollectionState) -> bool:
+        """Specifically Roc Wing, regardless of Roc being nerfed or not."""
+        return state.has(iname.roc_wing, self.player)
+
+    def has_jump_level_3(self, state: CollectionState) -> bool:
+        """Roc Wing and Double OR Kick Boots if Roc is nerfed. Otherwise, just Roc."""
+        if self.nerf_roc_wing:
+            return state.has(iname.roc_wing, self.player) and \
+                   state.has_any([iname.double, iname.kick_boots], self.player)
+        else:
+            return state.has(iname.roc_wing, self.player)
+
+    def has_jump_level_4(self, state: CollectionState) -> bool:
+        """Roc Wing and Kick Boots specifically if Roc is nerfed. Otherwise, just Roc."""
+        if self.nerf_roc_wing:
+            return state.has_all([iname.roc_wing, iname.kick_boots], self.player)
+        else:
+            return state.has(iname.roc_wing, self.player)
+
+    def has_jump_level_5(self, state: CollectionState) -> bool:
+        """Roc Wing, Double, AND Kick Boots if Roc is nerfed. Otherwise, just Roc."""
+        if self.nerf_roc_wing:
+            return state.has_all([iname.roc_wing, iname.double, iname.kick_boots], self.player)
+        else:
+            return state.has(iname.roc_wing, self.player)
+
+    def has_tackle(self, state: CollectionState) -> bool:
+        return state.has(iname.tackle, self.player)
+
+    def has_push(self, state: CollectionState) -> bool:
+        return state.has(iname.heavy_ring, self.player)
+
+    def has_kick(self, state: CollectionState) -> bool:
+        return state.has(iname.kick_boots, self.player)
+
+    def has_ice_or_stone(self, state: CollectionState) -> bool:
+        """Valid DSS combo that allows freezing or petrifying enemies to use as platforms."""
+        return state.has_any([iname.serpent, iname.cockatrice], self.player) and \
+            state.has_any([iname.mercury, iname.mars], self.player)
+
+    def can_touch_water(self, state: CollectionState) -> bool:
+        """Cleansing unless it's ignored, in which case this will always return True."""
+        if self.ignore_cleansing:
+            return True
+        return state.has(iname.cleansing, self.player)
+
+    def broke_iron_maidens(self, state: CollectionState) -> bool:
+        """Maiden Detonator unless the Iron Maidens start broken, in which case this will always return True."""
+        if self.iron_maiden_behavior == IronMaidenBehavior.option_start_broken:
+            return True
+        return state.has(iname.ironmaidens, self.player)
+
+    def can_open_ceremonial_door(self, state: CollectionState) -> bool:
+        """The required number of Last Keys unless it was set to 0, in which case this will always return True."""
+        if self.required_last_keys:
+            return state.has(iname.last_key, self.player, self.required_last_keys)
+        return True
```

Comment:
Will `state.has(iname.last_key, self.player, self.required_last_keys)` when `self.required_last_keys` equals 0 cause problems? If it doesn't, I feel like it could just be `return state.has(iname.last_key, self.player, self.required_last_keys)`.

Reply:
You're so right, that will just work.